dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -413,9 +424,16 @@ void ExchangeReceiverBase<RPCContext>::prepareMsgChannels()
 {
     if (enable_fine_grained_shuffle_flag)
         for (size_t i = 0; i < output_stream_count; ++i)
-            msg_channels.push_back(std::make_shared<ConcurrentIOQueue<std::shared_ptr<ReceivedMessage>>>(max_buffer_size));
+            msg_channels.push_back(std::make_shared<ConcurrentIOQueue<RecvMsgPtr>>(max_buffer_size));
     else
-        msg_channels.push_back(std::make_shared<ConcurrentIOQueue<std::shared_ptr<ReceivedMessage>>>(max_buffer_size));
+        msg_channels.push_back(std::make_shared<ConcurrentIOQueue<RecvMsgPtr>>(max_buffer_size));
+}
+
+template <typename RPCContext>
+void ExchangeReceiverBase<RPCContext>::prepareGRPCReceiveQueue()
+{
+    for (auto & msg_channel : msg_channels)
```

Comment:
how about adding `assert(!msg_channels.empty());`

Reply:
> how about adding `assert(!msg_channels.empty());`

I think it's unnecessary as `msg_channels` is impossible to be empty.